frontend/purposeful-frontend/components/Registration.js

```diff
@@ -14,11 +14,93 @@ import {
 } from "@chakra-ui/react";
 import { ViewIcon, ViewOffIcon } from "@chakra-ui/icons";
 import { Field, Form, Formik } from "formik";
+import fetchWrapper, { getAuthorities } from "../utils/fetch_wrapper";
+import notification from "../utils/notification";
+
+async function handleRegistrationForm(values, actions, GrantedAuth) {
+  const payload = {
+    email: values.email,
+    password: values.password,
+    firstname: values.firstname,
+    lastname: values.lastname,
+  };
+  let response = null;
+
+  if (
+    GrantedAuth.length === 0 &&
+    !window.location.pathname.includes("moderator")
+  ) {
+    // User is not authenticated and is registering as a regular user
+    response = await fetchWrapper(
+      "/api/appuser/regular",
+      null,
+      "POST",
+      payload
+    );
+  } else if (
+    GrantedAuth.includes("Owner") &&
+    window.location.pathname.includes("moderator")
+  ) {
+    // User is an admin and is registering a moderator
+    response = await fetchWrapper(
+      "/api/appuser/moderator",
+      null,
+      "POST",
+      payload
+    );
+  } else {
+    // User does not have the required permissions
```

Comment:
So in this else - if a user is logged in as anything else, they don't have permission. Hopefully we make the page endpoints controlled in such a way that this is never triggered? 

This led to a little bit of weird behavior when I was testing it. For example, someone who was logged in and trying to create an account was told they have no permission to access this endpoint. Maybe we will just prevent that with the routing between pages. 

We also get two error messages:

![image](https://user-images.githubusercontent.com/97904673/227745832-672c2468-7f0e-447f-ac0a-dea7963dd539.png)

This is my suggested fix. We make the frontend perform input validation checks, and display them as red error messages, and we use the popup for major errors only the backend would check

Reply:
Yes I will make the frontend do more input validation checks

When the endpoints are controlled, this kind of error (don't have permission) shouldn't occur. It is expected behavior though that a logged-in user cannot create an account

Reply:
The input validation in the frontend has been added